x-pack/plugins/task_manager/server/queries/oldest_idle_action_task.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ElasticsearchClient } from 'kibana/server';
+
+/**
+ * Returns the timestamp of the oldest action task that may still be executed.
+ * Useful for cleaning up related objects that may no longer be needed.
+ * @internal
+ */
+export const getOldestIdleActionTask = async (
+  client: Pick<ElasticsearchClient, 'search'>,
+  taskManagerIndex: string
+): Promise<number | string> => {
+  let oldestAlertAction: number | string = Date.now();
+  const response = await client.search<{ task: { runAt: string } }>(
+    {
+      size: 1,
+      index: taskManagerIndex,
+      body: {
+        sort: [{ 'task.runAt': { order: 'asc' } }],
+        query: {
+          bool: {
+            filter: {
+              bool: {
+                must: [
+                  {
+                    terms: {
+                      'task.taskType': [
+                        'actions:.email',
```

Comment:
This is obviously a bit fragile, for when we add new connector types.  The connector types are available via the actions plugin, but then we'll have a pre-req issue because the actions plugin depends on taskmanager.  

We can ship it this way for now, but should open an issue to do something less fragile later.

Or wait, have the plugins already registered?  Presumably task manager knows about all these tasks as well, and could then just filter on those matching `actions:*` (still fragile, in case some other plugin happens to register a task with that prefix, but ... seems better).

Reply:
> Or wait, have the plugins already registered? Presumably task manager knows about all these tasks as well, and could then just filter on those matching `actions:*` (still fragile, in case some other plugin happens to register a task with that prefix, but ... seems better).

Agreed that this is fragile. The plan is to tackle that as a follow up to keep the scope of changes here as small as possible. I have this included in the follow ups list in the PR description.